config.go

```diff
@@ -0,0 +1,14 @@
+package plugin
+
+// Configor is the interface plugins should implement in order to provide configuration interface to the user
+type Configor interface {
```

Comment:
Maybe this could be named `Configurer` to be consistent with the other plugin interfaces. 

Okay I think this is my last remark. Could you squash your commits?